# 例外

## 概要
SimpleORM4Jでは、内部的に例外が発生した場合は呼び出し元に対して以下の例外を送出します。

<table>
    <tr>
        <th>例外</th>
        <td>net.skuratani.simpleorm4j.So4jException</td>
    </tr>
</table>

SimpleORM4Jでは、内部で発生した例外のメッセージと原因をラップして例外を送出しますので、呼び出し元で以下の情報が必要な場合は参照をして、例外処理を実行してください

<table>
    <tr>
        <th>例外メッセージ</th>
        <td>So4jException#getMessage</td>
    </tr>
    <tr>
        <th>原因例外</th>
        <td>So4jException#getCause</td>
    </tr>
</table>
